numpy/core/records.py

```diff
@@ -673,8 +673,8 @@ def fromarrays(arrayList, dtype=None, shape=None, formats=None,
     _array = recarray(shape, descr)
 
     # populate the record array (makes a copy)
-    for i in range(len(arrayList)):
-        _array[_names[i]] = arrayList[i]
+    for name, item in zip(_names, arrayList):
+        _array[name] = item
```

Comment:
Could you merge the check in lines 667-671 into this loop so we iterate once over `arrayList`

Reply:
Makes sense, done.